lab_3_generate_by_ngrams/main.py

```diff
@@ -143,6 +195,14 @@ def _decode(self, corpus: tuple[int, ...]) -> Optional[tuple[str, ...]]:
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
+        if not isinstance(corpus, tuple) or len(corpus) == 0:
+            return None
+        decoded_text = []
+        for token in corpus:
+            if not self.get_token(token):
+                return None
+            decoded_text.append(self.get_token(token))
```

Comment:
call `self.get_token(token)` twice. Why?

Reply:
To check if it doesn't return None? What is the alternative?

Reply:
save to a variable?

Reply:
fixed